### ISSUE TITLE
How to handle nullable property references?

### ISSUE THREAD

**Issue by BenMorel** [zircote/swagger-php#670]
First of all sorry if I missed something obvious, but I've been struggling to find any information about how to handle this basic use case.

I have a Schema class that's not defined as `nullable`:

```php
/**
 * @OA\Schema()
 */
class Deal
{
}
```

When I want to use it in another Schema class and make it nullable, I write:

```php
/**
 * @OA\Schema()
 */
class Reservation
{
    /**
     * @OA\Property()
     *
     * @var Deal|null
     */
    public $deal;
}
```

However, `|null` is not taken into account like it is for scalar types, and this generates a `$ref` to the non-nullable Schema, like if I'd written `@var Deal`:

```yaml
components:
  schemas:
    Reservation:
      properties:
        deal:
          $ref: '#/components/schemas/Deal'
```

**Is this expected behaviour?** I'd expect this library to generate something like:

```yaml
        deal:
          nullable: true
          allOf:
            -
              $ref: '#/components/schemas/Deal'
```

Which is [what the spec seems to suggest](https://github.com/OAI/OpenAPI-Specification/issues/1368).

In the meantime, I tried to manually configure the `@Property`, with no luck so far:

```php
    /**
     * @OA\Property(allOf={@OA\Schema(ref="#/components/schemas/Deal")}, nullable=true)
     *
     * @var Deal|null
     */
    public $deal;
```

This does not work as it still generates the `$ref`:

```yaml
        deal:
          nullable: true
          allOf:
            -
              $ref: '#/components/schemas/Deal'
          $ref: '#/components/schemas/Deal'
```

**What did I miss?** Thanks in advance!

**Comment by BenMorel** [comment]
After digging through the code, it looks to me like this case is just not supported at the moment.
So I went ahead and implemented it in #671.

Feedback welcome, I desperately need this feature! 👍

**Comment by bfanger** [comment]
Closing this as "duplicate" of #671
